pkg/scheduler/internal/cache/snapshot_test.go

```diff
@@ -229,3 +230,107 @@ func TestCreateUsedPVCSet(t *testing.T) {
+					Status:     v1.NodeStatus{},
+				},
+			},
+			expectedNumNodes: 1,
+		},
+		{
+			name: "multiple nodes, pod with affinity",
+			pods: []*v1.Pod{
+				st.MakePod().Name("foo").Namespace("ns").Node("node-0").Obj(),
+				st.MakePod().Name("bar").Namespace("ns").PodAffinity("baz", &metav1.LabelSelector{MatchLabels: map[string]string{"baz": "qux"}}, st.PodAffinityWithRequiredReq).Node("node-2").Obj(),
+			},
+			nodes: []*v1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-0"},
+					Status:     v1.NodeStatus{},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-1"},
+					Status:     v1.NodeStatus{},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-2", Labels: map[string]string{"baz": "qux"}},
+					Status:     v1.NodeStatus{},
+				},
+			},
+			expectedNumNodes:         3,
+			expectedPodsWithAffinity: 1,
+		},
+		{
+			name: "multiple nodes, pod with affinity, pod with anti-affinity",
+			pods: []*v1.Pod{
+				st.MakePod().Name("foo").Namespace("ns").Volume(v1.Volume{}).Node("node-1").Obj(),
+				st.MakePod().Name("bar").Namespace("ns").PodAffinity("key", &metav1.LabelSelector{MatchLabels: map[string]string{"key": "value"}}, st.PodAffinityWithRequiredReq).Node("node-0").Obj(),
+				st.MakePod().Name("baz").Namespace("ns").PodAntiAffinity("another", &metav1.LabelSelector{MatchLabels: map[string]string{"another": "label"}}, st.PodAntiAffinityWithRequiredReq).Node("node-0").Obj(),
+			},
+			nodes: []*v1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-0", Labels: map[string]string{"key": "value"}},
+					Status:     v1.NodeStatus{},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-1", Labels: map[string]string{"another": "label"}},
+					Status:     v1.NodeStatus{},
+				},
+			},
+			expectedNumNodes:             2,
+			expectedPodsWithAffinity:     1,
+			expectedPodsWithAntiAffinity: 1,
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			snapshot := NewSnapshot(test.pods, test.nodes)
+			if len(test.nodes) == 0 {
+				info, err := snapshot.Get("dummy")
+				assert.Nil(t, info)
+				assert.Error(t, err)
+			} else {
+				info, err := snapshot.Get(test.nodes[0].Name)
+				assert.NotNil(t, info)
+				assert.NoError(t, err)
+			}
+
+			if len(test.pods) > 0 {
+				assert.False(t, snapshot.IsPVCUsedByPods(fmt.Sprintf("%s/%s", test.pods[0].Namespace, test.pods[0].Name)))
+			}
+
+			assert.Equal(t, snapshot.NumNodes(), test.expectedNumNodes)
+			list, err := snapshot.NodeInfos().List()
+			assert.Equal(t, len(list), test.expectedNumNodes)
+			assert.NoError(t, err)
+			affinityList, err := snapshot.HavePodsWithAffinityList()
+			assert.Equal(t, len(affinityList), test.expectedPodsWithAffinity)
+			assert.NoError(t, err)
+			antiAffinityList, err := snapshot.HavePodsWithRequiredAntiAffinityList()
+			assert.Equal(t, len(antiAffinityList), test.expectedPodsWithAntiAffinity)
+			assert.NoError(t, err)
```

Comment:
On more thing, we didn't use testify under scheduler package, for consistency, let's refactor like:
```
if test. expectedPodsWithAntiAffinity != len(antiaffinityList) {
   t.Errorf("unexpected antiAffinityList number, want: %v, got: %v", test.expectedPodsWithAntiAffinity, len(antiaffinityList))
}
```

same to others.